### PR TITLE
Fixes - Alteration Report

### PR DESCRIPTION
**Pull request by lekshmimallika-aot**
*Issue #:* /bcgov/entity#11096

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Review by severinbeauvais — APPROVED** [review]
Looks OK to me.

No unit tests to update?

**Review by vysakh-menon-aot — CHANGES_REQUESTED** [review]
I am not sure why we need this change. Based on `document_meta` service `Alteration Notice` is available when it is in `Completed` status. Is there a change in the requirement or is it a bug `core filing get_document_list`

File: legal-api/src/legal_api/reports/report.py [hunk 1]
```diff
@@ -374,12 +375,29 @@ def _format_alteration_data(self, filing):
...
+            new_legal_type = meta_data.get('alteration', {}).get('toLegalType')
+            prev_legal_name = meta_data.get('alteration', {}).get('fromLegalName')
+            to_legal_name = meta_data.get('alteration', {}).get('toLegalName')
+        else:
+            prev_legal_type = filing.get('business').get('legalType')
+            new_legal_type = filing.get('alteration').get('business').get('legalType')
+            prev_legal_name = filing.get('business').get('legalName')
+            identifier = filing.get('business').get('identifier')
+            name_request_json = filing.get('alteration').get('nameRequest')
+            to_legal_name = name_request_json.get('legalName', identifier[2:] + ' B.C. LTD.')
```
Comment (argush3) [inline legal-api/src/legal_api/reports/report.py:390]: I believe line 390 will error out if there is no `nameRequest` block in the alteration filing.

**Comment by lekshmimallika-aot** [comment]
> I am not sure why we need this change. Based on `document_meta` service `Alteration Notice` is available when it is in `Completed` status. Is there a change in the requirement or is it a bug in `core filing get_document_list`

Alteration Notice is available in the PAID status as well. If it is not returned in documents list it is a bug there

**Comment by lekshmimallika-aot** [comment]
> > I am not sure why we need this change. Based on `document_meta` service `Alteration Notice` is available when it is in `Completed` status. Is there a change in the requirement or is it a bug in `core filing get_document_list`
> 
> Alteration Notice is available in the PAID status as well. If it is not returned in documents list it is a bug there

Also this change is not mainly about that. Alteration report does not display legal type change correctly, company statement correctly even for a completed filing and lot of other things mentioned in the ticket

**Comment by vysakh-menon-aot** [comment]
> > > I am not sure why we need this change. Based on `document_meta` service `Alteration Notice` is available when it is in `Completed` status. Is there a change in the requirement or is it a bug in `core filing get_document_list`
> > 
> > 
> > Alteration Notice is available in the PAID status as well. If it is not returned in documents list it is a bug there
> 
> Also this change is not mainly about that. Alteration report does not display legal type change correctly, company statement correctly even for a completed filing and lot of other things mentioned in the ticket

Ok

**Review by argush3 — APPROVED** [review]

**Pull request merged**